validator/src/main/java/com/amazon/aoc/validators/TraceValidator.java

```diff
@@ -78,35 +80,56 @@ public void validate() throws Exception {
     // this method will hit get trace from x-ray service and get retrieved trace
     private Map<String, Object> getRetrievedTrace(List<String> traceIdList) throws Exception {
         Map<String, Object> flattenedJsonMapForRetrievedTrace = null;
+        List<Trace> retrieveTraceList = null;
+        int MAX_RETRY_COUNT = 3;
+        int count = 1;
+
+        while (retrieveTraceList == null || retrieveTraceList.isEmpty()) {
+            // wait for sample app to send trace data to x-ray
```

Comment:
you can throw a new baseexception("the trace id list is empty") then retry handler can capture it